src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/ISymbolExtensions.cs

```diff
@@ -364,13 +364,33 @@ public static bool IsParams([NotNullWhen(returnValue: true)] this ISymbol? symbo
         }
 
         public static ImmutableArray<IParameterSymbol> GetParameters(this ISymbol? symbol)
-            => symbol switch
+        {
+            return symbol switch
             {
                 IMethodSymbol m => m.Parameters,
-                IPropertySymbol nt => nt.Parameters,
+                IPropertySymbol p => p.Parameters,
+                INamedTypeSymbol nt when nt.IsRecord => GetRecordParameters(nt),
```

Comment:
yeah, i do not like this *at all*.  i'd say the caller needs to potentially deal with records.

Reply:
@CyrusNajmabadi Done. However, it became a little bit more complicated.